core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -1,78 +1,386 @@
+          .build();
 
   private final DynamoDbClient client;
   private final Optional<String> namespacePrefix;
   private final DynamoTableMetadataManager metadataManager;
+  private final ApplicationAutoScalingClient applicationAutoScalingClient;
 
   @Inject
   public DynamoAdmin(DynamoConfig config) {
+    AwsCredentialsProvider awsCredentialsProvider =
+        StaticCredentialsProvider.create(
+            AwsBasicCredentials.create(
+                config.getUsername().orElse(null), config.getPassword().orElse(null)));
+
     DynamoDbClientBuilder builder = DynamoDbClient.builder();
     config.getEndpointOverride().ifPresent(e -> builder.endpointOverride(URI.create(e)));
-    this.client =
+    client =
         builder
-            .credentialsProvider(
-                StaticCredentialsProvider.create(
-                    AwsBasicCredentials.create(
-                        config.getUsername().orElse(null), config.getPassword().orElse(null))))
+            .credentialsProvider(awsCredentialsProvider)
             .region(Region.of(config.getContactPoints().get(0)))
             .build();
+
+    ApplicationAutoScalingClientBuilder applicationAutoScalingClientBuilder =
+        ApplicationAutoScalingClient.builder();
+    config
+        .getEndpointOverride()
+        .ifPresent(e -> applicationAutoScalingClientBuilder.endpointOverride(URI.create(e)));
+    applicationAutoScalingClientBuilder.credentialsProvider(awsCredentialsProvider);
+    applicationAutoScalingClientBuilder.region(Region.of(config.getContactPoints().get(0)));
+    applicationAutoScalingClient = applicationAutoScalingClientBuilder.build();
+
     namespacePrefix = config.getNamespacePrefix();
     metadataManager = new DynamoTableMetadataManager(client, namespacePrefix);
   }
 
   @VisibleForTesting
   DynamoAdmin(DynamoTableMetadataManager metadataManager, Optional<String> namespacePrefix) {
     client = null;
+    applicationAutoScalingClient = null;
     this.metadataManager = metadataManager;
     this.namespacePrefix = namespacePrefix.map(n -> n + "_");
   }
 
+  @VisibleForTesting
+  DynamoAdmin(
+      DynamoDbClient dynamoDbClient,
+      ApplicationAutoScalingClient applicationAutoScalingClient,
+      DynamoTableMetadataManager metadataManager,
+      Optional<String> namespacePrefix) {
+    this.client = dynamoDbClient;
+    this.applicationAutoScalingClient = applicationAutoScalingClient;
+    this.metadataManager = metadataManager;
+    this.namespacePrefix = namespacePrefix;
+  }
+
   @Override
-  public void createNamespace(String namespace, Map<String, String> options)
-      throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+  public void createNamespace(String namespace, Map<String, String> options) {
+    // In Dynamo DB storage, namespace will be added to table name as prefix along with dot
+    // separator.
   }
 
   @Override
   public void createTable(
```

Comment:
I think this method is too big.
The code seems to have several logical units (for each component to build), so can you make them turn to methods and call them to make this method slim?

Reply:
@thongdk8 Somehow it is resolved without checked but was it difficult or it was unintentionally resolved by another change?

Reply:
Thank you. I did split the method to make it slimmer in [166ec54](https://github.com/scalar-labs/scalardb/commit/166ec546eb44ccd9bdc33932e43e2deabfb79cda).
This was marked as resolved because I did apply it, I mentioned [here](https://github.com/scalar-labs/scalardb/pull/301#discussion_r705922779). All the suggestions that were marked as resolved were applied.

Reply:
I meant, please reply back to all the comments. Otherwise, it's pretty hard to identify since there are many comments.

Reply:
I see. I'll do it from now on. Sorry about that.